torchvision/datasets/utils.py

```diff
@@ -42,14 +42,33 @@ def check_integrity(fpath: str, md5: Optional[str] = None) -> bool:
     return check_md5(fpath, md5)
 
 
-def download_url(url: str, root: str, filename: Optional[str] = None, md5: Optional[str] = None) -> None:
+def _get_redirect_url(url: str, hop_idx: int = 0, max_hops: int = 10) -> str:
+    import requests
```

Comment:
Nit: Currently we don't have this library in the dependencies, so it would be good to investigate if we can avoid using it.

Reply:
This is not the only place where `requests` is used:

https://github.com/pytorch/vision/blob/d0063f3d83beac01e85f3027c4de6499a8985469/torchvision/datasets/utils.py#L127-L137

I fully agree with the intention, but we probably should apply it to all places where `requests` is used.